src/components/Editor/HighlightLine.js

```diff
@@ -0,0 +1,104 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */
+
+// @flow
+import { Component } from "react";
+import { toEditorLine } from "../../utils/editor";
+import { getDocument } from "../../utils/editor/source-documents";
+import { isEqual } from "lodash";
+
+import { connect } from "react-redux";
+import { getVisibleSelectedFrame, getSelectedLocation } from "../../selectors";
+
+import type { Frame, Location } from "debugger-html";
+
+type Props = {
+  editor: Object,
+  selectedFrame: Frame,
+  selectedLocation: Location
+};
+
+export class DebugLine extends Component<Props> {
+  componentDidMount() {
+    const { selectedLocation, selectedFrame, editor } = this.props;
+    // this.setHighlightLine(selectedLocation, selectedFrame, editor);
+  }
+
+  shouldComponentUpdate(nextProps: Props) {
+    const { selectedLocation, selectedFrame } = this.props;
+
+    // Don't highlight the first selected location
+    if (!selectedLocation) {
+      return false;
+    }
+
+    // Only highlight selections :)
+    if (!nextProps.selectedLocation) {
+      return false;
+    }
+
+    // Don't highlight the same line twice
+    // NOTE: this protects us from highlights when the selected frame changes
+    if (nextProps.selectedLocation == selectedLocation) {
+      return false;
+    }
+
+    // Don't highlight the current debug line
+    if (this.isDebugLine(nextProps.selectedFrame, nextProps.selectedLocation)) {
+      return false;
+    }
+
+    return true;
+  }
```

Comment:
is it nice to have a verbose function or a more consolidated function?

Reply:
Why not both, 4 lines of comments - one for each condition above a consolidated if statement. Feels easier to read and fewer lines of code as well.